### PR TITLE
Add Addon.disable_all_files() and use it in force_disable()

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/13194

---

This patch makes sure we disable the files when we "force disable" an add-on.

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -506,6 +506,8 @@ def force_disable(self):
                  self.slug, amo.STATUS_DISABLED)
         self.update(status=amo.STATUS_DISABLED)
         self.update_version()
+        # See: https://github.com/mozilla/addons-server/issues/13194
+        self.disable_files()
 
     def force_enable(self):
         activity.log_create(amo.LOG.CHANGE_STATUS, self, amo.STATUS_APPROVED)
```
Comment (wagnerand) [inline src/olympia/addons/models.py]: Would it be safer to set the add-on status to `STATUS_INCOMPLETE` here?

Comment (eviljeff) [reply src/olympia/addons/models.py]: if it's `STATUS_INCOMPLETE` then we're not actually disabling the add-on, just disabling all the versions and leaving the developer free to upload more.

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -532,6 +534,10 @@ def allow_resubmission(self):
         log.info('Allow resubmission for addon "%s"', self.slug)
         DeniedGuid.objects.filter(guid=self.guid).delete()
 
+    def disable_files(self):
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: `disable_all_files`?

File: src/olympia/addons/models.py (hunk 1 above)
Comment (diox) [reply src/olympia/addons/models.py]: This is a comment on the `force_enable` method :)

File: src/olympia/addons/models.py [hunk 3]
```diff
@@ -506,6 +506,8 @@ def force_disable(self):
                  self.slug, amo.STATUS_DISABLED)
         self.update(status=amo.STATUS_DISABLED)
         self.update_version()
+        # See: https://github.com/mozilla/addons-server/issues/13194
+        self.disable_files()
 
     def force_enable(self):
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: I guess everyone is aware that `force_disable` won't be a reversible action any longer as `force_enable` won't re-enable the files?  (Really this change should have failed an integration test somewhere)

File: src/olympia/addons/models.py (hunk 1 above)
Comment (willdurand) [reply src/olympia/addons/models.py]: Let's file a new issue if this is a problem

Comment (eviljeff) [reply src/olympia/addons/models.py]: I didn't see these replies before I submitted my --edit :)

File: src/olympia/addons/models.py (hunk 3 above)
Comment (willdurand) [reply src/olympia/addons/models.py]: Yes, and that's why I added a comment with the link to the issue (it is described there)

File: src/olympia/addons/models.py (hunk 1 above)
Comment (wagnerand) [reply src/olympia/addons/models.py]: Done: https://github.com/mozilla/addons-server/issues/13746

**Pull request merged**